nat/traversal/config/parser.go

```diff
@@ -43,5 +43,9 @@ func (c *ConsumerConfigParser) Parse(config json.RawMessage) (ip string, port in
 	if err != nil {
 		return "", 0, "", errors.Wrap(err, "parsing consumer address:port failed")
 	}
-	return cfg.IP, cfg.Port, openvpn.ServiceType, nil
+
+	if cfg.IP == nil {
+		return "", 0, "", nil
```

Comment:
Why do you return empty data but with no error? I have no idea what is going on here - do you have some specific case in mind? Maybe you can add a test to show us what case are you covering here?

Reply:
If we return error here instead of empty data, we probably don't need to have `if port == 0 || IP == ""` in the upper layer.
```suggestion
		return "", 0, "", errors.New( "Remote party does not support NAT hole punching, IP:PORT is missing")
```

Reply:
good suggestion.